GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -690,73 +654,37 @@ public static string PushMultipleCmd(string remote, IEnumerable<GitPushAction> p
             }.ToString();
         }
 
-        public static string PushTagCmd(string path, string tag, bool all,
-            ForcePushOptions force = ForcePushOptions.DoNotForce)
+        public static string PushTagCmd(string path, string tag, bool all, ForcePushOptions force = ForcePushOptions.DoNotForce)
         {
-            path = path.ToPosixPath();
-
-            tag = tag.Replace(" ", "");
-
-            var sforce = GetForcePushArgument(force);
-
-            var sprogressOption = "";
-            if (VersionInUse.PushCanAskForProgress)
-            {
-                sprogressOption = "--progress ";
-            }
-
-            var options = string.Concat(sforce, sprogressOption);
-
-            if (all)
-            {
-                return "push " + options + "\"" + path.Trim() + "\" --tags";
-            }
-
-            if (!string.IsNullOrEmpty(tag))
-            {
-                return "push " + options + "\"" + path.Trim() + "\" tag " + tag;
-            }
-
-            return "";
-        }
-
-        public static string GetForcePushArgument(ForcePushOptions force)
-        {
-            var sforce = "";
-            if (force == ForcePushOptions.Force)
-            {
-                sforce = "-f ";
-            }
-            else if (force == ForcePushOptions.ForceWithLease)
+            var args = new ArgumentBuilder
             {
-                sforce = "--force-with-lease ";
-            }
+                "push",
+                force,
+                { VersionInUse.PushCanAskForProgress, "--progress" },
+                path.ToPosixPath().Trim().Quote(),
+                { all, "--tags" },
+                { !all, $"tag {tag.Replace(" ", "")}" }
+            };
 
-            return sforce;
+            return args.ToString();
```

Comment:
Not empty if (!string.IsNullOrEmpty(tag)) or !all

Reply:
I'm going to add this to the beginning of the method:

```c#
if (!all && string.IsNullOrWhiteSpace(tag))
{
    // TODO this is probably an error
    return "";
}
```